tests/ModelTest.php

```diff
@@ -1048,19 +1048,34 @@ public function testNumericFieldName(): void
         $this->assertEquals([3 => 'two.three'], $found[2]);
     }
 
-    public function testCreateOrFirst()
+    #[TestWith([true])]
+    #[TestWith([false])]
+    public function testCreateOrFirst(bool $transaction)
     {
+        $connection = DB::connection('mongodb');
+        $connection
+            ->getCollection('users')
+            ->createIndex(['email' => 1], ['unique' => true]);
+
+        if ($transaction) {
+            $connection->beginTransaction();
+        }
+
         Carbon::setTestNow('2010-06-22');
         $createdAt = Carbon::now()->getTimestamp();
+        $events = [];
+        self::registerModelEvents(User::class, $events);
         $user1 = User::createOrFirst(['email' => 'john.doe@example.com']);
 
         $this->assertSame('john.doe@example.com', $user1->email);
         $this->assertNull($user1->name);
         $this->assertTrue($user1->wasRecentlyCreated);
         $this->assertEquals($createdAt, $user1->created_at->getTimestamp());
         $this->assertEquals($createdAt, $user1->updated_at->getTimestamp());
+        $this->assertEquals(['saving', 'creating', 'created', 'saved'], $events);
 
         Carbon::setTestNow('2020-12-28');
+        $events = [];
```

Comment:
What's the point of calling `Carbon::setTestNow()` above? Is it related asserting `created_at` and `updated_at` still have the same value (i.e. the model was not modified)?

I suppose those assertions are consistent with expecting `$user2->wasRecentlyCreated` to still be false.